src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Composition;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    [Export(typeof(RequestExecutionQueue)), Shared]
```

Comment:
Ah, we'll also have multiple LSP servers long term anyway - the XAML LSP server is based off of this infrastructure, so we definitely need to make this per LSP server

Reply:
Good timing for this comment, because I was actually thinking this morning of moving the queue to the `AbstractRequestHandler` because with the work I did yesterday to pass the Document into the handler, and remove the SolutionProvider, it feels like a bit of a trap being on the server.

I'll have to think about it some more.

Reply:
> we want different execution queues per language server

This is currently imported by the language server, so we can just remove the `[Shared]` attribute right?